src/readbin/symbols.ml

```diff
@@ -66,4 +66,22 @@ let read ?demangle ~filename arch base  : string table =
             let mem = Memory.view ~from ~words base |> ok_exn in
             let name = demangle name in
             Table.add syms mem name |> ok_exn
-          with exn -> syms))
+          with _exn -> syms))
+
+let read_addrset filename : Addr.Hash_set.t =
+  let ic = In_channel.create filename in
+  let set_string = In_channel.input_all ic in
+  Addr.Hash_set.t_of_sexp @@ Sexp.of_string set_string
+
+let write_addrset ?filename (addr_set : Addr.Hash_set.t) : unit =
+  let oc = match filename with
+    | None -> Out_channel.stdout
+    | Some f -> Out_channel.create f in
+  Out_channel.output_string oc @@ Sexp.to_string (Addr.Hash_set.sexp_of_t addr_set);
+  Out_channel.close oc
+
+let write ?filename (syms : symbol table) : unit =
+  let fs_l = Table.foldi syms ~init:[] ~f:(fun mem _sym fs_list ->
+      let addr = Memory.min_addr mem in
+      addr::fs_list) in
+  write_addrset ?filename @@ Addr.Hash_set.of_list fs_l
```

Comment:
You shouldn't create an intermediate list for the only purpose to convert this list into the (Hash)Set. You can use `Hash_set.add` or better `Set.add` and add addresses directly to the set.